src/main/java/org/opentripplanner/middleware/models/MonitoredTrip.java

```diff
@@ -132,19 +144,29 @@
     public MonitoredTrip() {
     }
 
-    public MonitoredTrip(OtpDispatcherResponse otpDispatcherResponse) {
+    public MonitoredTrip(OtpDispatcherResponse otpDispatcherResponse) throws URISyntaxException {
         queryParams = otpDispatcherResponse.requestUri.getQuery();
         TripPlan plan = otpDispatcherResponse.getResponse().plan;
         itinerary = plan.itineraries.get(0);
-        initializeFromItinerary();
+
+        // extract trip time from parsed params and itinerary
+        initializeFromItineraryAndQueryParams();
     }
 
-    public void initializeFromItinerary() {
+    public void initializeFromItineraryAndQueryParams() throws IllegalArgumentException, URISyntaxException {
```

Comment:
Needs Javadoc.

Reply:
Added in 6fd00a976c063c5b46e7254a995fbbb628b90445